File: container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        bestArea = 0
        left = 0
        right = len(height) - 1
        
        while left < right:
            bestArea = max(bestArea, min(height[left], height[right])*(right-left))
            
            # move the pointer of the shorter side
            if height[left] > height[right]:
                right-=1
            else:
                left+=1
            
        return bestArea

File: BLIND-75/Array/container-with-most-water.ts
function maxArea(height: number[]): number {
  let max = 0;
  
  let left = 0;
  let right = height.length - 1;
  
  while (left < right) {
      max = Math.max(max, (right - left)*(Math.min(height[right], height[left])));
      
      if (height[left] < height[right]) {
          left++;
      } else {
          right--;
      }
  }
  
  return max;
};

// Start on opposite ends moving left or tright pointer when it is smaller, while comparing best area at all points till left and right converge
// then return the best area
